ui/pages/confirmations/components/confirm/info/shared/transaction-details.tsx

```diff
@@ -0,0 +1,54 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { isValidAddress } from 'ethereumjs-util';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowText,
+  ConfirmInfoRowUrl,
+} from '../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+import { useKnownMethodDataInTransaction } from '../../hooks';
+
+export const TransactionDetails = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
```

Comment:
Rather than casting at the point of use, would it be more reusable to define an alternate selector such as `useCurrentTransactionConfirmation` that returns the correct type?

Reply:
It's a good idea, but given that we need the generic selector in a few places like `useConfirmationNetworkInfo`, `syncConfirmPath`, MMI related components, and `useCurrentConfirmation` used in `<ConfirmTransaction />`, I believe keeping this casting where needed is as expressive while avoiding code triplication.